airlock_processor/StatusChangedQueueTrigger/__init__.py

```diff
@@ -1,18 +1,136 @@
+        if not result:
+            raise Exception("Failed parsing request properties")
+    except json.decoder.JSONDecodeError:
+        logging.error(f'Error decoding object: {body}')
+        raise
+    except Exception as e:
+        logging.error(f'Error extracting properties: {e}')
+        raise
+
+    return result
+
+
+def is_require_data_copy(new_status: str):
+    if new_status.lower() in ["submitted", "approved", "rejected", "blocked"]:
+        return True
+    return False
+
+
+def get_source_dest_env_vars(new_status: str, request_type: str, workspace_id: str):
+
+    # sanity
+    if is_require_data_copy(new_status) is False:
+        raise Exception("Given new status is not supported")
+
+    try:
+        tre_id = os.environ["TRE_ID"]
+        subscription_id = os.environ["AZURE_SUBSCRIPTION_ID"]
+    except KeyError as e:
+        logging.error(f'Missing environment variable: {e}')
+        raise
+
+    request_type = request_type.lower()
+    if request_type != "import" and request_type != "export":
+        raise Exception("Request type must be either import or export")
+
+    if new_status == 'submitted' and request_type == 'import':
+        source_account_name = "stalexim{}".format(tre_id)
+        dest_account_name = "stalipim{}".format(tre_id)
+        source_account_rg = "rg-{}".format(tre_id)
+        dest_account_rg = source_account_rg
+        logging.info("source account [%s rg: %s]. dest account [%s rg: %s]", source_account_name, source_account_rg, dest_account_name, dest_account_rg)
+    elif new_status == 'submitted' and request_type == 'export':
+        source_account_name = "stalexint{}".format(tre_id)
+        dest_account_name = "stalexip{}".format(tre_id)
+        source_account_rg = "rg-{}-ws-{}".format(tre_id, workspace_id)
+        dest_account_rg = source_account_rg
+        logging.info("source account [%s rg: %s]. dest account [%s rg: %s]", source_account_name, source_account_rg, dest_account_name, dest_account_rg)
+    elif new_status == 'approved' and request_type == 'import':
+        # https://github.com/microsoft/AzureTRE/issues/1841
+        pass
+    elif new_status == 'approved' and request_type == 'export':
+        # https://github.com/microsoft/AzureTRE/issues/1841
+        pass
+    elif new_status == 'rejected' and request_type == 'import':
+        # https://github.com/microsoft/AzureTRE/issues/1842
+        pass
+    elif new_status == 'rejected' and request_type == 'export':
+        # https://github.com/microsoft/AzureTRE/issues/1842
+        pass
+
+
+    if os.environ.get('ENABLE_LOCAL_DEBUG', NONE) is not NONE:
+        # Using the logged-in user
+        credential = DefaultAzureCredential()
+    else:
+        logging.info("using the Airlock processor's managed identity to get build storage management client")
+        credential = DefaultAzureCredential(managed_identity_client_id=os.environ["MANAGED_IDENTITY_CLIENT_ID"], exclude_shared_token_cache_credential=True)
```

Comment:
Why is there a difference in the auth mode?

Reply:
fixed